src/frequenz/reporting/_reporting.py

```diff
@@ -95,26 +95,46 @@ async def cumulative_energy(
             production += last_value_production
 
         else:
-            # Directly use energy values if using AC_ACTIVE_ENERGY
-            consumption = sum(
-                m2.value - m1.value
-                for m1, m2 in zip(metric_samples, metric_samples[1:])
-                if m2.value - m1.value > 0
-            )
-            production = sum(
-                m2.value - m1.value
-                for m1, m2 in zip(metric_samples, metric_samples[1:])
-                if m2.value - m1.value < 0
+            # Fetch energy consumption and production metrics separately
+            consumption_samples = [
+                sample
+                async for sample in client.list_microgrid_components_data(
+                    microgrid_components=[(microgrid_id, [component_id])],
+                    metrics=Metric.AC_ACTIVE_ENERGY_CONSUMED,
+                    start_dt=start_time,
+                    end_dt=end_time,
+                    resampling_period=resampling_period,
+                )
+            ]
+
+            production_samples = [
+                sample
+                async for sample in client.list_microgrid_components_data(
+                    microgrid_components=[(microgrid_id, [component_id])],
+                    metrics=Metric.AC_ACTIVE_ENERGY_DELIVERED,
+                    start_dt=start_time,
+                    end_dt=end_time,
+                    resampling_period=resampling_period,
+                )
+            ]
+
+            consumption = (
+                sum(
+                    m2.value - m1.value
+                    for m1, m2 in zip(consumption_samples, consumption_samples[1:])
```

Comment:
This should check that the diff is positive, because the energy can reset in between. Same for production

Reply:
Should I use something like `max(0, m2.value - m1.value)` to set it to 0 in case there is a reset?